scripts/newRecipe.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+# Exit on first failure
+set -e
+
+print_error() {
+    cat <<HELP
+Error: $1
+
+Usage:
+./scripts/newRecipe.sh <new recipe name> <optional recipe to copy>
+
+Examples:
+./scripts/newRecipe.sh MyNewRecipe
+./scripts/newRecipe.sh MyNewRecipe HelloHalogenHooks
+./scripts/newRecipe.sh MyNewRecipe recipes/HelloHalogenHooks/
+HELP
+exit 1
+}
+
+# Valid number of arguments is either 1 or 2
+[ $# -eq 0 ] && print_error 'No arguments'
+[ $# -gt 2 ] && print_error 'Too many arguments'
+
+# Use a default original recipe if one is not provided.
+# Just take basename of recipe path.
+orig=$(basename ${2:-HelloLog})
+
+echo "Creating new recipe $1 from $orig"
+
+# Echo all following script commands
+set -x
+
+# Copy original recipe to new recipe directory
+cp -r recipes/$orig recipes/$1
+
+# Replace all usages of the original recipe's name with the new recipe's name
+grep -rl $orig recipes/$1 | xargs sed -i "s/$orig/$1/g"
+
+# Build recipe
+make $1-build
```

Comment:
I'm not sure we want to build the recipe immediately after copying it... If the base includes additional dependencies that one does not want, it might affect the IDE's cache (i.e. what it considers to be included in the project when doing autocomplete and whatnot).
Other than that, this looks really good!